tests/Feature/Dashboard/TeamDownlineTest.php

```
@@ -0,0 +1,512 @@
+    public function test_level_totals_match_a_manual_sum_across_the_whole_tree(): void
+    {
+        [$root] = $this->threeGenerations();
+
+        $this->commission(agentId: 1, amount: 500, status: 'paid', paidAt: now()->subDays(2));
+        $this->commission(agentId: 2, amount: 300, status: 'paid', paidAt: now()->subDay());
+        $this->commission(agentId: 3, amount: 200, status: 'pending');
+        $this->commission(agentId: 4, amount: 150, status: 'paid', paidAt: now());
+        $this->commission(agentId: 5, amount: 75, status: 'pending');
+
+        $rollup = $this->service()->levelRollup($root, 30);
+        $rows = collect($rollup['rows'])->keyBy('depth');
```

PHPStan (2.2.8)

[error] Offset 'rows' does not exist on array<int, array<string, mixed>>.


(offsetAccess.notFound)